### ISSUE TITLE
[GPG][new trainer] Add support to new `GPG` method

### ISSUE THREAD

**Issue by lerogo** [huggingface/trl#3345]
### Method description

- **short description**: In this work, we revisit the traditional Policy Gradient (PG) mechanism and propose a minimalist RL approach termed Group Policy Gradient (GPG). Unlike conventional methods, GPG directly optimize the original RL objective, thus obviating the need for surrogate loss functions. **By eliminating the critic and reference models, avoiding KL divergence constraints, and addressing the advantage and gradient estimation bias**, our approach significantly simplifies the training process compared to Group Relative Policy Optimization (GRPO). Our approach achieves superior performance without relying on auxiliary techniques or adjustments.
- **paper link**: https://arxiv.org/abs/2504.02546

### Open source status

- [x] The method implementation is available
- [x] The model weights are available
- [x] The training datasets are available

### Provide useful links for the implementation

- source-code link: https://github.com/AMAP-ML/GPG
- model weights link: 
    - https://huggingface.co/GD-ML/Qwen2.5-Math-7B-GPG
- authors: 
    - GitHub: @cxxgtxy @lerogo @undyingjoker
    - Scholar: [Xiangxiang Chu](https://scholar.google.com/citations?user=jn21pUsAAAAJ), [Hailang Huang](https://scholar.google.com/citations?user=X0o0Ib8AAAAJ), [Xiao Zhang](https://github.com/undyingjoker), [Fei Wei](https://scholar.google.com/citations?user=xqrPe6gAAAAJ), [Yong Wang](https://www.semanticscholar.org/author/Yong-Wang/1683878) (AMAP, Alibaba Group)

**Comment by lerogo** [comment]
@qgallouedec Please check.

**Comment by qgallouedec** [comment]
Hi and thanks for proposing this addition.
I read quickly the paper, and it seems to me that the formulation of the objective rather close to GRPO in the case where $\beta=0$ and $\mu=1$:

```math
\mathcal{L}_{\text{GRPO}}(\theta) = -\frac{1}{\sum_{i=1}^G |o_i|} \sum_{i=1}^G \sum_{t=1}^{|o_i|} \left[ \frac{\pi_\theta(o_{i,t} \mid q, o_{i,< t})}{\left[\pi_\theta(o_{i,t} \mid q, o_{i,< t})\right]_{\text{no grad}}} \hat{A}_{i,t}  \right],
```


```math
\mathcal{L}_{\text{GPG}}(\theta) = -\frac{1}{\sum_{i=1}^{G} |o_i|} \sum_{i=1}^{G} \sum_{t=1}^{|o_i|} \left(\log \pi_{\theta}  (o_{i,t} \mid q, o_{i,< t})\hat{A}_{i,t}\right)
```

Can you confirm my calculations? If yes, then the difference in this case is that you don't divide by $\left[\pi_\theta(o_{i,t} \mid q, o_{i,< t})\right]_{\text{no grad}}$.

If I've understood everything correctly, I think it should be fairly simple to implement with a new possible value for `loss_type` in `GRPOConfig`. What do you think?

**Comment by lerogo** [comment]
Hi, thank you for your reply.

Your calculations are correct, and this part can be completed by modifying `GRPOConfig`. 

However, in our [latest GPG version](https://arxiv.org/pdf/2504.02546v3#page=3.30), we have addressed: 1. Reward bias; 2. Gradient estimation bias (Including gradient correction and resampling). This may not be achievable solely by modifying `GRPOConfig`.

**Comment by lerogo** [comment]
@qgallouedec Please check.
